src/Interfaces/Accessibility/AccessibilityInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types=1);
+
+namespace CodeFoundation\FlowConfig\Interfaces\Accessibility;
+
+interface AccessibilityInterface
+{
+    /**
+     * Gets whether the specified key can be retrieved.
+     *
+     * @param string $key
+     *
+     * @return bool
+     */
+    public function canGetKey(string $key): bool;
```

Comment:
I'd like some way of controlling access by type, so that if there were different entity types, it'd be possible to say that `TypeX` can set / get changes, but `TypeY` cannot.

Reply:
Would an additional argument of `?string $entityClass = null` be suited here, which will allow classes that implement the interface to see the full class name of the entity, or do you think providing the entity instance itself would be useful (say if you were wanting to return based on the value of a method or property within that entity)?

Reply:
I'd go with the actual entity class. That's how it's been done for `getByEntity()` and `setByEntity()`, and that automatically exposes any information needed.

I can see some potential cases where an implementation of this class may do something like look up ACLs to see which users are allowed to have a settings, and which aren't for example.